### PR TITLE
Sign tool: Added command line option to save enclave hash to a file

### PR DESCRIPTION
**Pull request by omeg**
Since there seems to be no convenient way to get enclave's MRENCLAVE value during the build process, this patch adds such option to the signing tool (-hash <hash file path>).

Signed-off-by: Rafal Wojdyla <omeg@invisiblethingslab.com>

### CODE REVIEW

**Comment by andyzyb** [comment]
Will review this enhancement, thanks.

**Comment by rootkovska** [comment]
I'd just like to point out that knowing the value of MRENCLAVE at the build time is sometimes of critical importance. Such a situation occurs e.g. if one builds a system comprising 2 or more SGX enclaves, and wants to authenticate other enclaves via their hash (so MRENCLAVE value), as opposed to the value of MRSIGNER.

This is especially important when attempting to create audit-able systems where the customer does not need to be forced to fully trust the enclave's vendor.

**Comment by rodolfoams** [comment]
This is great! Already using it in my projects. Thanks for this contribution, @omeg.

**Comment by andyzyb** [comment]
We are working a major release. This patch likely will be merged after that release in a few weeks.

File: sdk/sign_tool/SignTool/sign_tool.cpp [hunk 1]
```diff
@@ -1183,6 +1188,22 @@ int main(int argc, char* argv[])
...
+        FILE *fp = fopen(path[HASH], "wb");
+        if(fp == NULL)
+        {
+            se_trace(SE_TRACE_ERROR, OVERALL_ERROR);
+            goto clear_return;
+        }
+        if(fwrite(enclave_hash, sizeof(enclave_hash), 1, fp) != 1)
+        {
+            se_trace(SE_TRACE_ERROR, OVERALL_ERROR);
+            goto clear_return;
```
Comment (lzha101) [inline sdk/sign_tool/SignTool/sign_tool.cpp]: Need `fclose(fp)` before clear_return

**Comment by omeg** [comment]
Added requested change.

**Comment by lzha101** [comment]
In the latest 1.9 release, we've added one command  `dump` and one option `-dumpfile` to save necessary metadata info to a file including enclave hash. Please refer to latest signing tool implementation and related documentations for this feature.

**Comment by gmatesunny** [comment]
Could someone tell me(kind of manual) how to use it in SDK 2.0

**Comment by lzha101** [comment]
Please refer to  [Developer Reference](https://download.01.org/intel-sgx/linux-2.1.2/docs/Intel_SGX_Developer_Reference_Linux_2.1.2_Open_Source.pdf) , section `Enclave Signing Tool` for how to use this feature.

**Comment by andyzyb** [comment]
Implemented in the latest release of signing tool.

**Pull request closed without merging**